### ISSUE TITLE
Improve accessibility of admin pagination

### ISSUE THREAD

**Issue by dacook** [openfoodfoundation/openfoodnetwork#13331]
The new pagination implementation (currently used only on Admin > Products - Bulk Edit screen) does not use the ideal HTML which means assistive and other technologies cannot use it. Improving the markup can aid users and made the code more maintainable.

* Arising from discussion on https://github.com/openfoodfoundation/openfoodnetwork/issues/13288#issuecomment-2864231480

## Technical Requirements
1. Wrap it in a `<nav>` or if that's not possible the wrapper div should have a role="navigation"

2. Add `aria-label="Pagination"` on the `<nav>` (no need to add "navigation" as screen readers will add that context from the nav tag). 
3. Pagelist should be an `<ul>` with `<li>`s
4. Pagelist items should be links not buttons _// (a link takes you to a different page, a button is used for interactivity on a page)_
5. Add `aria-current="page"` on the `<a>` tag of the current page
6. Page links should have a meaningful label so either add `aria-label="page"` or if that's tricky for internationalisation then you can add a visually hidden `<span>` inside the link:

```html
<a href="path/to/page"><span class="visuallyhidden">Page</span> 1</a>
```
7. The next and previous buttons should have a `rel="next"` and "prev" attribute

### Development notes
All user-facing text (eg `aria-label` contents) should be translatable. See https://github.com/openfoodfoundation/openfoodnetwork/wiki/Internationalisation-(i18n) for more info.

The template can be found here:
https://github.com/openfoodfoundation/openfoodnetwork/blob/master/app/views/admin/shared/_stimulus_pagination.html.haml

## Great resource
The GOV.UK Design System has a pretty [neat guide on pagination](https://design-system.service.gov.uk/components/pagination/) with most of the pattern/design questions answered (what to do on first page, last page, between pages, changing stuff for smaller screens etc.)

Credit: thanks @krisztin for providing these notes!

**Comment by dacook** [comment]
@krisztin , I copied your notes here. Regarding translation of `aria-` attributes, that shouldn't be a problem so we will go with that for cleaner html. 
Just one question, for the page links, you mentioned `aria-label="page"`, which doesn't include the page number. I'm assuming that's because the link text ("1") would be announced next, therefore adding it to the aria-label would result in unnecessary repetition; is that right?

**Comment by krisztin** [comment]
Hey @dacook , sorry, my write up wasn't super clear on that bit, left out the page # from the label. Page link should be, for instance:

```html
<a href="path/to/page" aria-label="Page 1">1</a>
```

**Comment by dacook** [comment]
Thanks, updated the description.

**Comment by RachL** [comment]
@dacook I've labeled this one "front-end" hard on the welcome new developers board, is that correct?

**Comment by marincarroll** [comment]
Can I take this issue?

**Comment by sigmundpetersen** [comment]
Sure thank you @marincarroll 👍 
I've assigned you. And welcome 🙂

**Comment by marincarroll** [comment]
> Sure thank you [@marincarroll](https://github.com/marincarroll) 👍 I've assigned you. And welcome 🙂

Thanks for the warm welcome @sigmundpetersen. I think this is in a good place to PR - it resembles and behaves like the GOV.UK example, I tested it with VoiceOver on Mac and that seems to work well. https://github.com/openfoodfoundation/openfoodnetwork/compare/master...marincarroll:openfoodnetwork:improve-accessibility-of-admin-pagination

I also added ARIA labels to the Prev/Next button which just had an icon before and weren't read out. And I removed the (hidden) disabled Prev/Next buttons to match the UK.GOV example.

Quick question as I (unfortunately) don't have experience writing tests, since this is a modification of an existing feature is there an opportunity for a new test? It seems like the  suite is already failing some tests on master

**Comment by sigmundpetersen** [comment]
Hi Marin.
Please submit your PR and it can be reviewd by our team 👍 It will then also be run against our CI suite here on GH Actions.
If you have added funcitonality I think it should be covered by a new test, yes.

**Comment by dacook** [comment]
Hi Marin and welcome! The tests do generally success on master on GitHub actions (although it's true some can perform differently when running on a development environment). I went ahead and took a look at one of the failing specs on your branch and it is related to your branch and will need fixing.
```
Failure/Error: expect(page).not_to have_content "Previous"
       expected not to find text "Previous" in "Previous 1 2 Next"
```

In this case, you mentioned that "Previous" is not necessary when disabled, so you can simply delete that test line.
If you need more help fixing it, you can reach out in the slack #dev channel, where you have the best chance of someone being online at the same time as you.

**Comment by marincarroll** [comment]
> Hi Marin and welcome! The tests do generally success on master on GitHub actions (although it's true some can perform differently when running on a development environment). I went ahead and took a look at one of the failing specs on your branch and it is related to your branch and will need fixing.
> 
> ```
> Failure/Error: expect(page).not_to have_content "Previous"
>        expected not to find text "Previous" in "Previous 1 2 Next"
> ```
> 
> In this case, you mentioned that "Previous" is not necessary when disabled, so you can simply delete that test line. If you need more help fixing it, you can reach out in the slack #dev channel, where you have the best chance of someone being online at the same time as you.

Thanks for your help! Will check this out.

**Comment by marincarroll** [comment]
Ok got tests passing, it was because of crossover with areas that use what I assume is the "old" pagination (/admin/shared/_pagy_links.html.haml_). Just to confirm, I am not supposed to make any changes to the "old" version as part of this ticket? Was unsure since "Improve accessibility of admin pagination" could refer to the areas it's used as well (Enterprises, Users, Zones)

> Hi Marin and welcome! The tests do generally success on master on GitHub actions (although it's true some can perform differently when running on a development environment). I went ahead and took a look at one of the failing specs on your branch and it is related to your branch and will need fixing.
> 
> ```
> Failure/Error: expect(page).not_to have_content "Previous"
>        expected not to find text "Previous" in "Previous 1 2 Next"
> ```
> 
> In this case, you mentioned that "Previous" is not necessary when disabled, so you can simply delete that test line. If you need more help fixing it, you can reach out in the slack #dev channel, where you have the best chance of someone being online at the same time as you.
